### PR TITLE
WRO-6267: Added unit tests to starter theme components

### PR DESCRIPTION
**Pull request by vladut-clapou-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Added unit tests to starter theme components

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-6267

### Comments
Enact-DCO-1.0-Signed-off-by: Clapou George-Vladut(vladut.clapou@lgepartner.com)

### CODE REVIEW

File: packages/theme/template/Checkbox/tests/Checkbox-specs.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	});
+
+
+	test('should check the checkbox with one click', () => {
+		render(<Checkbox data-testid="checkbox" />);
+
+		const checkboxItem = screen.getByTestId('checkbox');
+		const expected = 'selected';
+
+		fireEvent.click(checkboxItem);
```
Comment (daniel-stoian-lgp) [inline packages/theme/template/Checkbox/tests/Checkbox-specs.js:29]: can you use userEvent.click?

File: packages/theme/template/Checkbox/tests/Checkbox-specs.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	});
+
+	test('should uncheck the checkbox with two clicks', () => {
+		render(<Checkbox data-testid="checkbox" />);
+
+		const actual = screen.getByTestId('checkbox');
+		const expected = 'selected';
+
+		fireEvent.click(actual);
+		fireEvent.click(actual);
```
Comment (daniel-stoian-lgp) [inline packages/theme/template/Checkbox/tests/Checkbox-specs.js:41]: can you use userEvent.click?

File: packages/theme/template/CheckboxItem/tests/CheckboxItem-specs.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		expect(element).toBeInTheDocument();
+	});
+
+	test('should check with click', () => {
+		render(<CheckboxItem>Hello CheckboxItem</CheckboxItem>);
+
+		const checkboxItem = screen.getAllByRole('checkbox');
+		const expected = 'true';
+
+		fireEvent.click(checkboxItem[0]);
```
Comment (daniel-stoian-lgp) [inline packages/theme/template/CheckboxItem/tests/CheckboxItem-specs.js:22]: can you use userEvent.click?

File: packages/theme/template/CheckboxItem/tests/CheckboxItem-specs.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	});
+
+	test('should uncheck with 2 click', () => {
+		render(<CheckboxItem>Hello CheckboxItem</CheckboxItem>);
+
+		const checkboxItem = screen.getAllByRole('checkbox');
+		const expected = 'false';
+
+		fireEvent.click(checkboxItem[0]);
+		fireEvent.click(checkboxItem[0]);
```
Comment (daniel-stoian-lgp) [inline packages/theme/template/CheckboxItem/tests/CheckboxItem-specs.js:35]: can you use userEvent.click?

File: packages/theme/template/package.json [hunk 5]
```diff
@@ -28,19 +28,19 @@
     "dist/*"
   ],
   "dependencies": {
-    "@enact/core": "^4.0.0",
-    "@enact/i18n": "^4.0.0",
-    "@enact/spotlight": "^4.0.0",
-    "@enact/ui": "^4.0.0",
+    "@enact/core": "^4.5.0-alpha.2",
```
Comment (daniel-stoian-lgp) [inline packages/theme/template/package.json]: let's use the new beta versions

File: packages/theme/template/package.json [hunk 6]
```diff
@@ -28,19 +28,19 @@
...
+    "@enact/i18n": "^4.5.0-alpha.2",
+    "@enact/spotlight": "^4.5.0-alpha.2",
+    "@enact/ui": "^4.5.0-alpha.2",
     "classnames": "^2.2.6",
-    "prop-types": "^15.6.2",
-    "ramda": "^0.25.0",
-    "react": "^17.0.1"
+    "prop-types": "^15.8.1",
+    "ramda": "^0.28.0",
+    "react": "^18.0.0"
```
Comment (daniel-stoian-lgp) [inline packages/theme/template/package.json]:
```suggestion
    "react": "^18.1.0"
```

File: .travis.yml [hunk 7]
```diff
@@ -5,5 +5,14 @@ node_js:
...
-    - npm run lint
+    - echo -e "\x1b\x5b35;1m*** Starting eslint...\x1b\x5b0m"
+    - npm run lint 
+    - echo -e "\x1b\x5b35;1m*** eslint complete\x1b\x5b0m"
+    - cd ./packages/theme/template
+    - echo -e "\x1b\x5b35;1m*** Installing dependencies... \x1b\x5b0m"
+    - npm install
+    - echo -e "\x1b\x5b35;1m*** Starting tests...\x1b\x5b0m"
+    - npm run test 
+    - echo -e "\x1b\x5b35;1m*** Tests complete\x1b\x5b0m"
```
Comment (daniel-stoian-lgp) [inline .travis.yml]:
```suggestion
    - echo -e "\x1b\x5b35;1m*** Tests complete\x1b\x5b0m"
    
```

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM. Good job!

File: packages/theme/template/BodyText/tests/BodyText-specs.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import '@testing-library/jest-dom';
+import {render, screen} from '@testing-library/react';
+
+import BodyText, {BodyTextBase} from '../BodyText';
+import {Cell} from '@enact/ui/Layout';
```
Comment (sjro) [inline packages/theme/template/BodyText/tests/BodyText-specs.js]: Reorder this line to be the first line

Comment (sjro) [inline packages/theme/template/BodyText/tests/BodyText-specs.js]: Reorder this line to be the first line

**Review by sjro — APPROVED** [review]

File: packages/theme/template/package.json [hunk 9]
```diff
@@ -28,19 +28,19 @@
...
   "dependencies": {
-    "@enact/core": "^4.0.0",
-    "@enact/i18n": "^4.0.0",
-    "@enact/spotlight": "^4.0.0",
-    "@enact/ui": "^4.0.0",
+    "@enact/core": "^4.5.0-beta.1",
+    "@enact/i18n": "^4.5.0-beta.1",
+    "@enact/spotlight": "^4.5.0-beta.1",
+    "@enact/ui": "^4.5.0-beta.1",
     "classnames": "^2.2.6",
```
Comment (MikyungKim) [inline packages/theme/template/package.json]: ^2.3.1?

File: packages/theme/template/Checkbox/tests/Checkbox-specs.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+		expect(checkboxElement).not.toHaveClass('selected');
+	});
+
+	test('should include the selected class when selected', () => {
+		render(<CheckboxBase selected />);
+		const checkboxElement = screen.getByRole('checkbox');
+
+		expect(checkboxElement).toHaveClass('selected');
+	});
+
```
Comment (MikyungKim) [inline packages/theme/template/Checkbox/tests/Checkbox-specs.js:21]: Please remove the empty line.

File: packages/theme/template/Checkbox/tests/Checkbox-specs.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+	test('should check the checkbox with one click', () => {
+		render(<Checkbox data-testid="checkbox" />);
+
+		const checkboxItem = screen.getByTestId('checkbox');
```
Comment (MikyungKim) [inline packages/theme/template/Checkbox/tests/Checkbox-specs.js]: Do we need to use `getByTestId` here? Why don't we use `getByRole`?

File: packages/theme/template/Checkbox/tests/Checkbox-specs.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+		fireEvent.click(checkboxItem);
+
+		expect(checkboxItem).toHaveClass(expected);
+	});
+
+	test('should uncheck the checkbox with two clicks', () => {
+		render(<Checkbox data-testid="checkbox" />);
+
+		const actual = screen.getByTestId('checkbox');
```
Comment (MikyungKim) [inline packages/theme/template/Checkbox/tests/Checkbox-specs.js]: Same here.

**Comment by MikyungKim** [comment]
@vladut-clapou-lgp , I'd like to run lint for "templates" folder in travis but I couldn't do it :( would you please make this work?

**Comment by vladut-clapou-lgp** [comment]
> @vladut-clapou-lgp , I'd like to run lint for "templates" folder in travis but I couldn't do it :( would you please make this work?

Hi @MikyungKim. I managed to make this work. Can you please have a look? Thanks!

**Review by MikyungKim — APPROVED** [review]
:lgtm:

**Pull request merged**